### PR TITLE
Update Mustache to v2.2.1

### PR DESCRIPTION
**Pull request by petetnt**
This PR updates Mustache to version `v2.2.1` from `v0.7`. 

Mostly everything Brackets uses Mustache for works as it used to except for `Mustache.compile` which has been completely removed for over 2 years. This PR also provides a simple polyfill for it for not to break extensions that might still be using it.

### CODE REVIEW

**Comment by ficristo** [comment]
I'm all for use up-to-date libraries, but is there any reason beyond that to update `Mustache`?
Did you look if there is any extension use `Mustache.compile`?
Maybe using something like https://github.com/dangoor/BracketsExtensionGrabber.

File: src/thirdparty/CodeMirror [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 61ca2db598f3f1d79e837f326bc740e8bc49e62a
+Subproject commit 291b06eb891ff63bc010795bd492e9a672c13566
```
Comment (ficristo) [inline src/thirdparty/CodeMirror]: I don't think this change belong in this PR

Comment (petetnt) [inline src/thirdparty/CodeMirror]: Fixed

**Comment by petetnt** [comment]
@ficristo No much other reason than it's a low hanging fruit (that I had already done when doing #12006) that could provide some performance benefits, especially for ExtensionManager if the parsing is faster (which it should be).

File: src/brackets.js [hunk 2]
```diff
@@ -114,6 +114,23 @@ define(function (require, exports, module) {
             return CodeMirror;
         }
     });
+    
+    // DEPRECATED: Mustache.compile has been deprecated since Mustache 0.8,
+    // but for now we use a polyfill for it so as to avoid breaking extensions in the
+    // interim.
+    Object.defineProperty(window.Mustache, "compile", {
```
Comment (MiguelCastillo) [inline src/brackets.js]: This is great. I am not sure how prominent `compile` is, but it is good to warn and tag it a deprecated.  @ficristo is right about letting extension authors this change.  @MarcelGerber might be able to give us some guidance on this. :)

**Comment by MarcelGerber** [comment]
I had a look at updating Mustache some months ago as well, but when I did it, I faced some display issues in the Extension Manager.
So we should definitely look out for regressions.

I may get to have a look soon (at least I hope so).

**Comment by MarcelGerber** [comment]
I've now had a look at this PR, and noticed the issue I had before is still present - in the Extension Manager dialog, extension author's names are no longer shown.

(If you need help in resolving the merge conflicts, please say so, I have a working branch local)

**Comment by MarcelGerber** [comment]
My latest changes are up at `MarcelGerber:update-mustache` (see https://github.com/adobe/brackets/compare/8093791728cbff6f7734fbad41882481c82087f9...MarcelGerber:update-mustache for a changeset against current master).

I've also filed the issue with Mustache: https://github.com/janl/mustache.js/issues/566

And, to complete, I've looked for extensions that use `Mustache.compile` and found these:
* brackets-cuwire
* brackets-wallpaper
* genuitec.team-extensions (for some reason, they have a copy of `ExtensionManagerView`?)
* shreder *uses `Mustache.compilePartial`*, for which we don't have a polyfill!

**Comment by petetnt** [comment]
Blarg, screwed rebasing this PR with your changes. @MarcelGerber feel free to open a new PR with your branch. Really great work finding the issues as always, I'll check if there's a simple way to polyfill the the `compilePartial` method.

**Pull request closed without merging**